apps/web-giddh/src/app/services/inventory.service.ts

```diff
@@ -1477,16 +1478,107 @@ export class InventoryService {
-        return this.http.post(apiUrl,request.body);
+        let apiUrl = this.config.apiUrl + INVENTORY_API.GET_CUSTOMER_VENDOR_DISCOUNT_USERS?.replace(':companyUniqueName', encodeURIComponent(this.companyUniqueName))?.replace(':page', encodeURIComponent(request.page))?.replace(':count', encodeURIComponent(request.count))?.replace(':group', encodeURIComponent(request.group))?.replace(':query', encodeURIComponent(request.query || ''));
+
+        return this.http.get(apiUrl);
+    }
+
+    /**
+     * Get Customer/Vendor Wise Discount All Customer and Group list
+     *
+     * @param {*} request
+     * @return {*}  {Observable<BaseResponse<any, any>>}
+     * @memberof InventoryService
+     */
+    public getFlattenAccountsList(request: any): Observable<BaseResponse<any, any>> {
+        this.companyUniqueName = this.generalService.companyUniqueName;
+        let apiUrl = this.config.apiUrl + INVENTORY_API.GET_FLATTEN_ACCOUNTS?.replace(':companyUniqueName', encodeURIComponent(this.companyUniqueName))?.replace(':page', encodeURIComponent(request.page))?.replace(':count', encodeURIComponent(request.count))?.replace(':group', encodeURIComponent(request.group))?.replace(':query', encodeURIComponent(request.query || ''));
+
+        return this.http.get(apiUrl);
+    }
+
+    /**
+    * Get Customer/Vendor Wise Discount Customer list
+    *
+    * @param {*} request
+    * @return {*}  {Observable<BaseResponse<any, any>>}
+    * @memberof InventoryService
+    */
+    public getStockList(request: any): Observable<BaseResponse<any, any>> {
+        this.companyUniqueName = this.generalService.companyUniqueName;
+        let apiUrl = this.config.apiUrl + INVENTORY_API.GET_ALL_STOCKS?.replace(':companyUniqueName', encodeURIComponent(this.companyUniqueName))?.replace(':page', encodeURIComponent(request.page))?.replace(':count', encodeURIComponent(request.count))?.replace(':query', encodeURIComponent(request.query || ''));
+
+        return this.http.get(apiUrl);
+    }
+
+    /**
+     * Get list of stock and its variant based on user (uniqueName)
+     *
+     * @param {*} request
+     * @return {*}  {Observable<BaseResponse<any, any>>}
+     * @memberof InventoryService
+     */
+    public getAllDiscount(request: any): Observable<BaseResponse<any, any>> {
+        this.companyUniqueName = this.generalService.companyUniqueName;
+        let apiUrl = this.config.apiUrl + INVENTORY_API.GET_ALL_DISCOUNTS?.replace(':companyUniqueName', encodeURIComponent(this.companyUniqueName))?.replace(':page', encodeURIComponent(request.page))?.replace(':count', encodeURIComponent(request.count))?.replace(':uniqueName', encodeURIComponent(request.uniqueName))?.replace(':query', encodeURIComponent(request.query || ''));
+
+        return this.http.get(apiUrl);
+    }
+
+    public deleteDiscountRecord(request: any): Observable<BaseResponse<any, any>> {
```

Comment:
add documentation on all methods you added

Reply:
done